### PR TITLE
Fix tsconfig resolution for navigation

### LINKED ISSUE #192772: tsconfig.json references navigation doesn't understand folder references
Navigating to the valid `./src` goes to `<folder>/src.json`, not the expected `<folder>/src/tsconfig.json`

Image

### PR DESCRIPTION
**Pull request by ronakj**
Fixes #192772

Tested following scenarios -
1. Relative path without json at the end
2. Relative path with json at the end
3. tsconfig which resolves to a module in `node_modules`

Tested on Linux & Windows

### CODE REVIEW

**Comment by ronakj** [comment]
@mjbvz can you take a look and approve these changes it looks ok?

**Review by ezhil56x — APPROVED** [review]
Works as expected!

File: extensions/typescript-language-features/src/languageFeatures/tsconfig.ts [hunk 1]
```diff
@@ -154,7 +154,7 @@ async function getTsconfigPath(baseDirUri: vscode.Uri, pathValue: string): Promi
 			return absolutePath;
 		}
 		return absolutePath.with({
-			path: `${absolutePath.path}.json`
+			path: `${absolutePath.path}/tsconfig.json`
```
Comment (mjbvz) [inline extensions/typescript-language-features/src/languageFeatures/tsconfig.ts:157]: From https://github.com/microsoft/TypeScript/blob/febfd442cdba343771f478cf433b0892f213ad2f/src/compiler/commandLineParser.ts#L3005 I think we need to check both `PATH.json` and `PATH/tsconfig.json`, at least when resolving the path in `extends`

Comment (ronakj) [reply extensions/typescript-language-features/src/languageFeatures/tsconfig.ts]: Updated. From my own testing and looking at typescript documentation, extends and references are different in resolution. I have added a `linkType` parameter to the navigation command to resolve this, though we can also create a separate command if necessary.

**Review by mjbvz — APPROVED** [review]

**Review by lramos15 — APPROVED** [review]

**Comment by mjbvz** [comment]
Thanks @ronakj! This will be in the next VS code insiders build

**Pull request merged**